### PR TITLE
Add polling support.

### PR DESCRIPTION
**Pull request by georgebrock**
This PR updates the action to poll for the provisioning status after the initial request. This lays the groundwork for reporting more useful information to the user, e.g. a definitive success message, or debugging information on a failure.

### CODE REVIEW

File: test/action_test.rb [hunk 1]
```diff
@@ -31,9 +36,54 @@ def test_immediate_success
...
+        "GITHUB_REF" => "main",
+        "GITHUB_REPOSITORY" => "monalisa/smile",
+        "GITHUB_SHA" => "abcdef1234567890",
+        "GITHUB_TOKEN" => "my-very-secret-token",
+        "INPUT_REGIONS" => "WestUs2",
+        "INPUT_SKU_NAME" => "futuristicQuantumComputer",
+      },
+      job_ids: [job_id],
+      status_responses: {
+        job_id => ["pending", "running", "reticulating-splines", "complete"]
```
Comment (dmilburn) [inline test/action_test.rb:60]: <3

File: cache-codespace.sh [hunk 2]
```diff
@@ -1,5 +1,31 @@
...
+
+poll_status () {
+  local job_id="$1"
+  local attempt="${2:-1}"
+
+  get_status "$job_id"
+
+  echo $status
+
+  if [[ "$status" == *complete* ]]; then
```
Comment (dmilburn) [inline cache-codespace.sh:18]: Is "complete" here signaling the status of the actual `JobStatus` object? How were you thinking about signaling the success vs. failure different behavior of the PrebuildTemplate itself? Would that happen in a different endpoint?

Comment (georgebrock) [reply cache-codespace.sh:18]: `complete` is what our endpoint returns when the process is 100% complete:

- either the `JobStatus.state` is `success` and there is no `PrebuildTemplate` record; or
- the `JobStatus.state` is `success`, and there is a `PrebuildTemplate` record, and VSCS reports the environment for the `PrebuildTemplate` is in the `archived` state.

The logic for this isn't merged yet, but it is defined here: https://github.com/github/github/pull/187914/files#diff-f3179e3a9c506a2eb6dd7637700f686cc0ea3106e1f5d65e54a9f262a2be2a01R23-R34

Comment (dmilburn) [reply cache-codespace.sh:18]: I think this is also missing a third possibility where `JobStatus.state` is `error` if there was an error from vscs when creating the PrebuildTemplate then I think the action should appear failed here as well

Comment (georgebrock) [reply cache-codespace.sh:18]: Yep, I was thinking that handling the failure state was out-of-scope for now until the logging endpoint is ready.

**Review by dmilburn — APPROVED** [review]
LGTM!

**Review by josebalius — APPROVED** [review]
This is great ✨

**Pull request merged**